clang/lib/Driver/ToolChains/ClassicFlang.cpp

```diff
@@ -1054,37 +1054,50 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
   }
 
   // Add vscale range
-  unsigned vscaleMin = 1U;
-  unsigned vscaleMax = 16U;
-  bool hasVscaleRange = false;
+  unsigned vscaleMin = 0;
+  unsigned vscaleMax = 0;
+  bool hasSVE = false;
   if (Arg *A = Args.getLastArg(options::OPT_msve_vector_bits_EQ)) {
     StringRef Val = A->getValue();
-
-    if (Val.equals("scalable"))
-      hasVscaleRange = true;
-    else {
-      unsigned bits = (std::stoul(Val.str()) >> 7U);
-
-      if ((bits < vscaleMin) || (bits > vscaleMax)) {
-        D.Diag(diag::warn_drv_clang_unsupported) << A->getAsString(Args);
-        hasVscaleRange = false;
-      } else {
-        vscaleMin = vscaleMax = bits;
-        hasVscaleRange = true;
+    if (Val.equals("128") || Val.equals("256") || Val.equals("512") ||
+        Val.equals("1024") || Val.equals("2048") || Val.equals("128+") ||
+        Val.equals("256+") || Val.equals("512+") || Val.equals("1024+") ||
+        Val.equals("2048+")) {
+      unsigned Bits = 0;
+      if (Val.endswith("+"))
+        Val = Val.substr(0, Val.size() - 1);
+      else {
+        bool Invalid = Val.getAsInteger(10, Bits); (void)Invalid;
+        assert(!Invalid && "Failed to parse value");
```

Comment:
Why not put the function call inside the `assert`? e.g.
```c++
assert(!Val.getAsInteger(10, Bits) && "Failed to parse value");
```

Reply:
That's OK. But I just keep the format in Clang.

Reply:
OK.